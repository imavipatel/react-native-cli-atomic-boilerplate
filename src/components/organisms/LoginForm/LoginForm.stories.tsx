import * as React from 'react';
import { storiesOf } from '@storybook/react-native';
import { LoginForm } from './LoginForm';
import CenterView from '../../../../storybook/utils/CenterView/CenterView';

storiesOf('LoginForm', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('default', () => (
    <LoginForm onSubmit={() => {}} title={'Login Form title'} />
  ));
