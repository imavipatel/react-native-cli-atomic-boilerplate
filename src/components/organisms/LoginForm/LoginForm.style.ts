import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flex: 4,
    justifyContent: 'space-between',
    flexDirection: 'column',
    padding: 30,
  },
});
