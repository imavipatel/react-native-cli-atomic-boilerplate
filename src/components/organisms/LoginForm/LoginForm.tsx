import React from 'react';
import { View } from 'react-native';

import { Button, PressableLabel } from '../../atoms';
import { Input } from '../../molecules';
import { styles } from './LoginForm.style';

interface LoginFormProps {
  onSubmit: () => void;
  onClickLabel: () => void;
  title: string;
}

export const LoginForm: React.FC<LoginFormProps> = props => {
  return (
    <View style={styles.container}>
      <Input label="Email" keyboardType="email-address" testID="email-input" />
      <Input label="Password" secureTextEntry testID="password-input" />
      <Button onPress={props.onSubmit} text="Sign in" testID="submit-button" />
      <PressableLabel onPress={props.onClickLabel} testID="registration-label">
        Don't have an account? Sign up
      </PressableLabel>
    </View>
  );
};
