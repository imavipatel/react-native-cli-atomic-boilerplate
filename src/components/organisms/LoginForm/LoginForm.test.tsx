import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { LoginForm } from './LoginForm';

describe('LoginForm', () => {
  const spyOnSubmit = jest.fn();
  const spyOnClickLabel = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const tree = render(
      <LoginForm
        onClickLabel={spyOnClickLabel}
        onSubmit={spyOnSubmit}
        title={'Login Form title'}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should call the provided onSubmit function when pressed', () => {
    const component = (
      <LoginForm
        onClickLabel={spyOnClickLabel}
        onSubmit={spyOnSubmit}
        title={'Login Form title'}
      />
    );
    const wrapper = render(component);

    fireEvent(wrapper.getByTestId('submit-button'), 'onPress');
    expect(spyOnSubmit).toHaveBeenCalledTimes(1);
  });

  it('should call the provided onClickLabel function when pressed', () => {
    const component = (
      <LoginForm
        onClickLabel={spyOnClickLabel}
        onSubmit={spyOnSubmit}
        title={'Login Form title'}
      />
    );
    const wrapper = render(component);

    fireEvent(wrapper.getByTestId('registration-label'), 'onPress');
    expect(spyOnClickLabel).toHaveBeenCalledTimes(1);
  });
});
