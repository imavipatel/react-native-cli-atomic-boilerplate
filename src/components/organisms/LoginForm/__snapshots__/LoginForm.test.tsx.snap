// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 4,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "padding": 30,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "height": 78,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
      testID="label"
    >
      Email
    </Text>
    <TextInput
      keyboardType="email-address"
      placeholder="Email"
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#08949B",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 1,
          "fontSize": 16,
          "fontWeight": "600",
          "justifyContent": "center",
          "minHeight": 50,
          "minWidth": 200,
          "paddingLeft": 17,
          "paddingRight": 17,
        }
      }
      testID="email-input"
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "height": 78,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
      testID="label"
    >
      Password
    </Text>
    <TextInput
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#08949B",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 1,
          "fontSize": 16,
          "fontWeight": "600",
          "justifyContent": "center",
          "minHeight": 50,
          "minWidth": 200,
          "paddingLeft": 17,
          "paddingRight": 17,
        }
      }
      testID="password-input"
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#08949B",
        "borderRadius": 20,
        "borderStyle": "solid",
        "borderWidth": 0,
        "justifyContent": "center",
        "minHeight": 60,
        "minWidth": 200,
      }
    }
    testID="submit-button"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
      testID="button-text"
    >
      Sign in
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="registration-label"
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
      testID="label"
    >
      Don't have an account? Sign up
    </Text>
  </View>
</View>
`;
